src/client.rs

```diff
@@ -568,3 +573,32 @@ pub extern "C" fn rustls_client_session_write_tls(
         rustls_result::Ok
     }
 }
+
+/// Register callbacks for persistence of TLS session data. This means either
+/// session IDs (TLSv1.2) or . Both
+/// keys and values are highly sensitive data, containing enough information
+/// to break the security of the sessions involved.
+///
```

Comment:
This should document something about the lifetime of the `userdata` pointer. I.e. if it's non-NULL, the pointed-to data must live at least as long as the config object and any sessions created from that config object.